ownCloud/Client/Actions/Actions+Extensions/MarkupAction.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  MarkupAction.swift
+//  ownCloud
+//
+//  Created by Matthias Hühne on 21/01/2020.
+//  Copyright © 2020 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2020, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import ownCloudSDK
+
+@available(iOS 13.0, *)
+class MarkupAction : Action {
+	override class var identifier : OCExtensionIdentifier? { return OCExtensionIdentifier("com.owncloud.action.markup") }
+	override class var category : ActionCategory? { return .normal }
+	override class var name : String? { return "Markup".localized }
+	override class var keyCommand : String? { return "E" }
+	override class var keyModifierFlags: UIKeyModifierFlags? { return [.command] }
+	override class var locations : [OCExtensionLocationIdentifier]? { return [.moreItem, .moreFolder, .keyboardShortcut] }
+
+	var interactionControllerDispatchGroup : DispatchGroup?
+	var interactionController : EditDocumentViewController?
+
+	// MARK: - Extension matching
+	override class func applicablePosition(forContext: ActionContext) -> ActionPosition {
+		let supportedMimeTypes = ["image", "pdf"]
+		if forContext.items.contains(where: {$0.type == .collection}) {
```

Comment:
@hosy could be simplified I think:
- return .none by default at the end
- check if single item is passed AND has supported mimeType -> return .middle